tests/validators/blocks/test_grand_calculated_summary_block_validator.py

```diff
@@ -125,3 +125,38 @@ def test_invalid_non_repeating_grand_calculated_summary_referencing_repeating_ca
     errors = validator.validate()
 
     assert errors == expected_error_messages
+
+
+def test_invalid_repeating_grand_calculated_summary_with_static_calculated_summary_with_repeating_answers_same_list():
```

Comment:
Not anything blocking and just a detail (I can't find much otherwise, the code looks good) but it is the longest test title I've seen, it even breaks the Python's standard 79 characters a line rule so maybe use abbreviations for gcs and cs?

Reply:
I've got other review comments asking for consistency between abbreviations and full names, so I want to keep it consistent but I have reworded it slightly to make it shorter since the doc string makes it clear anyway